### ISSUE TITLE
Apple TV integration fails after Apple TV restarts - Suggests reconfiguration

### ISSUE THREAD

**Issue by mikesalz** [postlund/pyatv#1043]
**Describe the bug**
Copying issue from https://github.com/home-assistant/core/issues/48901 because it seems to have gotten lost there.

I have 3 Apple TV units. Two are 4th generation, and one is 4k. With all 3 of them, I encounter errors when I restart the Apple TV unit. The integration stops working, and HA suggests to reconfigure the Apple TV that was restarted.

This happens about 50-75% of the time. Restarting HA fixes the issue. But the errors always come back after 1 or 2 restarts of the Apple TV.
![image](https://user-images.githubusercontent.com/46855061/116313996-855a4100-a77c-11eb-8e95-925f90b3d245.png)

```log
Logger: homeassistant.components.apple_tv
Source: components/apple_tv/init.py:330
Integration: Apple TV (documentation, issues)
First occurred: 8:12:58 PM (2 occurrences)
Last logged: 8:13:06 PM

Failed to connect

Traceback (most recent call last): File
“/usr/src/homeassistant/homeassistant/components/apple_tv/init.py”, line 246, in _connect_loop await self._connect(conf) File “/usr/src/homeassistant/homeassistant/components/apple_tv/init.py”, line 330, in _connect self.atv = await connect(conf, self.hass.loop, session=session) File “/usr/local/lib/python3.8/site-packages/pyatv/init.py”, line 76, in connect await atv.connect() File “/usr/local/lib/python3.8/site-packages/pyatv/mrp/init.py”, line 703, in connect await self._protocol.start() File “/usr/local/lib/python3.8/site-packages/pyatv/mrp/protocol.py”, line 103, in start self.device_info = await self.send_and_receive(msg) File “/usr/local/lib/python3.8/site-packages/pyatv/mrp/protocol.py”, line 180, in send_and_receive return await self._receive(identifier, timeout) File “/usr/local/lib/python3.8/site-packages/pyatv/mrp/protocol.py”, line 188, in _receive await asyncio.wait_for(semaphore.acquire(), timeout) File “/usr/local/lib/python3.8/asyncio/tasks.py”, line 501, in wait_for raise exceptions.TimeoutError() asyncio.exceptions.TimeoutError


Logger: pyatv.mrp.protocol
Source: /usr/local/lib/python3.8/site-packages/pyatv/mrp/protocol.py:44
First occurred: 8:12:53 PM (2 occurrences)
Last logged: 8:12:53 PM
```
**To Reproduce**
Restart the Apple TV 2-3 times, and it should happen.

**Expected behavior**
Not this

**System Setup (please complete the following information):**
 - OS: Home Assistant Docker Container on a Synology NAS
 - Python: Whatever the current version of Home Assistant uses
 - pyatv: Whatever the current version of Home Assistant uses
- Apple TV: 4K and HD.  All on the latest TVOS

**Additional context**

**Comment by postlund** [comment]
I will try to reproduce this later tonight if I get the time, thanks for reporting!

**Comment by mikesalz** [comment]
Thank you @postlund ! Please let me know if there is anything I can do to help.

**Comment by mikesalz** [comment]
Hi @postlund!  Just wondering if you have had any luck here, or if there is anything I can do to help debug?

**Comment by postlund** [comment]
@mikesalz Sorry, I forgot about this. I'm gonna try to reproduce as soon as I get some time.

**Comment by mikesalz** [comment]
Thanks @postlund !  I actually have an update for you.  I recently upgraded my two Apple TV HD units to the 2021 4k model.  Since that time, HA has stopped asking me to reconfigure each time the Apple TV is restarted.  However, I do still receive the heartbeat errors in the log each time:

> Logger: pyatv.mrp.protocol
> Source: /usr/local/lib/python3.8/site-packages/pyatv/mrp/protocol.py:44
> First occurred: June 16, 2021, 11:46:36 PM (1 occurrences)
> Last logged: June 16, 2021, 11:46:36 PM
> 
> heartbeat 381 failed after 2 tries

**Comment by postlund** [comment]
@mikesalz That's interesting! 381 failed attempts isn't that long, just a couple of minutes. Is it always around that time? Or at least a low value? I would love to know what happens and if you happen to own a mac, maybe you can help me out by installing Xcode, making your unit into a dev unit and extract the log from the mediaremoted process in System Messages? Would help me a lot as I'm probably doing something wrong.

**Comment by mikesalz** [comment]
I'm not sure if it is always around the same time.  But I restarted one of the Apple TVs just now in order to trigger the error again:

> Logger: pyatv.mrp.protocol
> Source: /usr/local/lib/python3.8/site-packages/pyatv/mrp/protocol.py:44
> First occurred: June 16, 2021, 11:46:36 PM (2 occurrences)
> Last logged: 12:24:15 PM
> 
> heartbeat 381 failed after 2 tries
> heartbeat 451 failed after 2 tries

I do have a Mac and I do have XCode installed, although I am not very familiar with it.  Can you walk me through the steps you'd like me to take?

**Comment by postlund** [comment]
I will try, it's been a while 😊 You can try if this works:

https://stackoverflow.com/questions/44382841/how-do-you-perform-wireless-debugging-in-xcode-9-with-ios-11-apple-tv-4k-etc/44383502#44383502

Assuming it does, you should start the "System Messages" app on your Mac, hopefully your device will appear to the left. From there it should be fairly easy to start streaming events and filtering on mediaremoted as process (up to the right).

**Comment by mikesalz** [comment]
I tried, but I think I hit a dead end.  I have XCode, but when I connected to the new Apple TV 4k it said I needed the latest version.  So I went to the app store, and it turns out I need to be on MacOS 11 in order to install it.  But my Mac is too old and I can't update to anything newer than Catalina (10.15.7).  I wish they'd hurry up and release the new MX/M2 iMacs!  I'm waiting on that before I get a new computer.
